src/cljs/commiteth/common.cljs

```diff
@@ -12,16 +12,24 @@
                    :on-change #(reset! val-ratom (-> % .-target .-value))})]))
 
 (defn dropdown [props title val-ratom items]
-  (fn []
-    (if (= 1 (count items))
-      (reset! val-ratom (first items)))
-    [:select.ui.basic.selection.dropdown
-     (merge props {:on-change
-                   #(reset! val-ratom (-> % .-target .-value))})
-     (doall (for [item items]
-              ^{:key item} [:option
-                            {:value item}
-                            item]))]))
+  "If val-ratom is set, preselect it in the dropdown.
+   Add value of val-ratom if it's missing from items list.
```

Comment:
Should we allow the value of the val-ratom to not be in the item coll? Wouldn't it be better to just require it, since the item coll tells us the order? Not sure if this requires changes in other places of the app though, in which case we shouldn't probably do it.

Reply:
This component is not used anywhere else, so things won't break if we change something here.

Why would we want to allow `val-ratom` not to be present in `items`? Because if e.g. MetaMask account is switched to one different from currently set in SOB, `items` will not contain it, hence user will not be able to select their current account in the dropdown.

Or do you suggest to properly populate `items` at the invocation point of `dropdown` instead?

Reply:
Yeah, I meant that creating the whole collection of available items should probably be done outside, so that we don't need special behaviours/conditions inside, that is, just a coll of items and optionally one of those values as selected.

Reply:
Moved items construction to invocation point. Dropdown preselects either the atom value or title.